README.md

```diff
@@ -68,6 +71,78 @@ func main() {
 }
 ```
 
+Users who require access to individual partitions can use the partitioned mode which exposes access to partition-level
+consumers:
+
+```go
+package main
+
+import (
+  "fmt"
+  "log"
+  "os"
+  "os/signal"
+
+  cluster "github.com/bsm/sarama-cluster"
+)
+
+func main() {
+
+	// init (custom) config, enable errors and notifications
+	// set mode to ConsumerModePartitions
+	config := cluster.NewConfig()
+	config.Consumer.Return.Errors = true
+	config.Group.Return.Notifications = true
+	config.Group.Mode = cluster.ConsumerModePartitions
+
+	// init consumer
+	brokers := []string{"127.0.0.1:9092"}
+	topics := []string{"my_topic", "other_topic"}
+	consumer, err := cluster.NewConsumer(brokers, "my-consumer-group", topics, config)
+	if err != nil {
+		panic(err)
+	}
+	defer consumer.Close()
+
+	// trap SIGINT to trigger a shutdown.
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, os.Interrupt)
+
+	// consume partitions, watch notifications
+	for {
+		select {
+		case part, more := <-consumer.Partitions():
+			if !more {
+				return
+			}
+
+			// start a separate goroutine to consume the partition;
+			go func(pc cluster.PartitionConsumer) {
+				select {
+				case msg, more := <-pc.Messages():
+					if !more {	// exit when the partition is closed
+						return
+					}
+					fmt.Fprintf(os.Stdout, "%s/%d/%d\t%s\t%s\n", msg.Topic, msg.Partition, msg.Offset, msg.Key, msg.Value)
+					consumer.MarkOffset(msg, "")	// mark message as processed
+				case err, more := <-consumer.Errors():
+					if !more {	// exit when the partition is closed
+						return
+					}
+					log.Printf("Error: %s\n", err.Error())
+				}
+			}(part)
+		case ntf, more := <-consumer.Notifications():
```

Comment:
I know it's just an example, but what's the advantage of putting everything in a giant select?

I think it'd be equally useful to `for range` loop Notifications in another goroutine... after all it should close once the consumer does.. makes the example more concise and less indented.

Reply:
Modified example: https://gist.github.com/crast/2f179d88d169cea7e5deadb7f11d593f

Reply:
well, preferences, but I agree, the giant select is not pretty